### PR TITLE
Fix several bugs

### PR DESCRIPTION
**Pull request by adonisgarciac**
### What does this PR do?

Fixes:

1.- Conditional export user roles if there are at least one.
2.- Export users inside a users directory
3.- Export organization name for inventories and not export org id
4.- Export role parameter as "role" and not "roles" because roles is wrong
5.- Export WFJT, project and credentials for user and teams roles.
6.- Add state present in export WFJT
7.- Fix object diff for credential_types when is tower and not aap.
8.- Use controller_oauthtoken in all object_diff.

### How should this be tested?

Launch test playbooks for filetree_create, filetree_read and object_diff roles.

### CODE REVIEW

File: roles/filetree_create/tasks/users.yml [hunk 1]
```diff
@@ -22,20 +22,20 @@
...
   vars:
-    __path: "{{ output_path }}/{{ current_user_dir | regex_replace('/', '_') }}"
+    __path: "{{ output_path }}/{{ current_user_dir | regex_replace('/', '_') }}/users"
   loop: "{{ current_users | selectattr('organizations', 'defined') | map(attribute='organizations') | flatten | unique }}"
   loop_control:
     loop_var: current_user_dir
     label: "{{ __path }}"
 
-- name: "Add current users to the output yaml file: {{ output_path }}/<ORGANIZATION_NAME>/current_users_<USERNAME>.yaml"
+- name: "Add current users to the output yaml file: {{ output_path }}/<ORGANIZATION_NAME>/users/current_users_<USERNAME>.yaml"
```
Comment (ivarmu) [inline roles/filetree_create/tasks/users.yml]: We can remove `current_users_` prefix for the generated files, as they are already placed into `users/` directory

File: roles/filetree_create/tasks/users.yml [hunk 2]
```diff
@@ -22,20 +22,20 @@
...
   ansible.builtin.template:
     src: "templates/current_users.j2"
     dest: "{{ __dest }}"
     mode: '0644'
   vars:
     current_users_asset_value: "{{ current_user_dir.0 }}"
-    __dest: "{{ output_path }}/{{ current_user_dir.1 | regex_replace('/', '_') }}/current_users_{{ current_user_dir.0.username | regex_replace('/', '_') }}.yaml"
+    __dest: "{{ output_path }}/{{ current_user_dir.1 | regex_replace('/', '_') }}/users/current_users_{{ current_user_dir.0.username | regex_replace('/', '_') }}.yaml"
```
Comment (ivarmu) [inline roles/filetree_create/tasks/users.yml]: Idem here.

File: roles/object_diff/tasks/groups.yml [hunk 3]
```diff
@@ -13,16 +13,14 @@
                                               'has_inventory_sources': 'false',
                                               'not__total_groups': '0',
                                               'not__kind': 'smart'},
-                                            host=controller_hostname, username=controller_username,
-                                            password=controller_password, verify_ssl=controller_validate_certs)
+                                            host=controller_hostname, oauth_token=controller_oauthtoken, verify_ssl=controller_validate_certs)
```
Comment (ivarmu) [inline roles/object_diff/tasks/groups.yml:16]: I prefer to maintain this in two lines, for both better reading and avoid linting problems.

File: roles/object_diff/tasks/groups.yml [hunk 4]
```diff
@@ -13,16 +13,14 @@
...
                                    }}"
 
 - name: "OBJECT DIFF: Get the API list of all groups in the inventories at organization {{ orgs }}"
   ansible.builtin.set_fact:
     __controller_api_groups: "{{ (__controller_api_groups | default([])) + query(controller_api_plugin, 'groups',
                                           query_params={'inventory': current_inventory.id},
-                                          host=controller_hostname, username=controller_username,
-                                          password=controller_password, verify_ssl=controller_validate_certs)
+                                          host=controller_hostname, oauth_token=controller_oauthtoken, verify_ssl=controller_validate_certs)
```
Comment (ivarmu) [inline roles/object_diff/tasks/groups.yml:23]: I prefer to maintain this in two lines, for both better reading and avoid linting problems.

File: roles/object_diff/tasks/hosts.yml [hunk 5]
```diff
@@ -21,8 +21,7 @@
   ansible.builtin.set_fact:
     __controller_api_hosts: "{{ (__controller_api_hosts | default([])) + query(controller_api_plugin, 'hosts',
                                     query_params={'inventory': current_inventory.id},
-                                    host=controller_hostname, username=controller_username,
-                                    password=controller_password, verify_ssl=controller_validate_certs)
+                                    host=controller_hostname, oauth_token=controller_oauthtoken, verify_ssl=controller_validate_certs)
```
Comment (ivarmu) [inline roles/object_diff/tasks/hosts.yml:24]: I prefer to maintain this in two lines, for both better reading and avoid linting problems.

File: roles/object_diff/tasks/roles.yml [hunk 6]
```diff
@@ -12,8 +12,7 @@
     - name: "OBJECT DIFF: Get the API list of all roles"
       ansible.builtin.set_fact:
         __controller_api_roles: "{{ query(controller_api_plugin, 'roles',
-                                          host=controller_hostname, username=controller_username,
-                                          password=controller_password, verify_ssl=controller_validate_certs)
+                                          host=controller_hostname, oauth_token=controller_oauthtoken, verify_ssl=controller_validate_certs)
```
Comment (ivarmu) [inline roles/object_diff/tasks/roles.yml:17]: I prefer to maintain this in two lines, for both better reading and avoid linting problems.

File: roles/object_diff/tasks/main.yml [hunk 7]
```diff
@@ -27,6 +27,12 @@
   tags:
     - always
 
+- name: "Check if the connection is to an Ansible Tower or to Automation Platform"
+  ansible.builtin.set_fact:
+    is_aap: "{{ lookup(controller_api_plugin, 'ping', host=controller_hostname, oauth_token=controller_oauthtoken, verify_ssl=controller_validate_certs).version is version('4.0.0', '>=') }}"
```
Comment (ivarmu) [inline roles/object_diff/tasks/main.yml]: Could you split this line in more lines?

**Review by ivarmu — CHANGES_REQUESTED** [review]

**Comment by ivarmu** [comment]
Please, fix the conflicts and the requested changes so that PR can be merged.

Great job @adonisgarciac !!! Thanks!

File: roles/object_diff/tasks/groups.yml (hunk 3 above)
Comment (adonisgarciac) [reply roles/object_diff/tasks/groups.yml:16]: It is not to remove one line, it is to change user/pass access for oauth_token which is already in the rest of tasks and there, they are in one line. Example:

https://github.com/redhat-cop/controller_configuration/blob/devel/roles/object_diff/tasks/inventories.yml#L6

File: roles/object_diff/tasks/groups.yml (hunk 4 above)
Comment (adonisgarciac) [reply roles/object_diff/tasks/groups.yml:23]: It is not to remove one line, it is to change user/pass access for oauth_token which is already in the rest of tasks and there, they are in one line. Example:

https://github.com/redhat-cop/controller_configuration/blob/devel/roles/object_diff/tasks/inventories.yml#L6

File: roles/object_diff/tasks/hosts.yml (hunk 5 above)
Comment (adonisgarciac) [reply roles/object_diff/tasks/hosts.yml:24]: It is not to remove one line, it is to change user/pass access for oauth_token which is already in the rest of tasks and there, they are in one line. Example:

https://github.com/redhat-cop/controller_configuration/blob/devel/roles/object_diff/tasks/inventories.yml#L6

File: roles/object_diff/tasks/roles.yml (hunk 6 above)
Comment (adonisgarciac) [reply roles/object_diff/tasks/roles.yml:17]: It is not to remove one line, it is to change user/pass access for oauth_token which is already in the rest of tasks and there, they are in one line. Example:

https://github.com/redhat-cop/controller_configuration/blob/devel/roles/object_diff/tasks/inventories.yml#L6

File: roles/object_diff/tasks/main.yml (hunk 7 above)
Comment (adonisgarciac) [reply roles/object_diff/tasks/main.yml]: I copied from https://github.com/redhat-cop/controller_configuration/blob/devel/roles/filetree_create/tasks/all.yml#L4

i will change both

File: roles/object_diff/tasks/groups.yml (hunk 3 above)
Comment (ivarmu) [reply roles/object_diff/tasks/groups.yml:16]: perfect then :-)

File: roles/object_diff/tasks/groups.yml (hunk 4 above)
Comment (ivarmu) [reply roles/object_diff/tasks/groups.yml:23]: perfect then :-)

File: roles/object_diff/tasks/hosts.yml (hunk 5 above)
Comment (ivarmu) [reply roles/object_diff/tasks/hosts.yml:24]: perfect then :-)

File: roles/object_diff/tasks/main.yml (hunk 7 above)
Comment (ivarmu) [reply roles/object_diff/tasks/main.yml]: Even better! Thanks!

File: roles/object_diff/tasks/roles.yml (hunk 6 above)
Comment (ivarmu) [reply roles/object_diff/tasks/roles.yml:17]: perfect then :-)

**Review by ivarmu — APPROVED** [review]
LGTM. Thanks for the good job!!!

**Pull request merged**